tests/conftest.py

```
@@ -0,0 +1,98 @@
+from collections.abc import Callable, Iterable
+from typing import Any
+from unittest.mock import MagicMock, Mock
+
+import pytest
+import streamlit as st
+
+
+class DummySessionState(dict):
+    """Mock session state that supports both dict and attribute access."""
+
+    def __getattr__(self, key: str) -> Any:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `__getattr__`

(ANN401)